ocpmodels/preprocessing/atoms_to_graphs.py

```diff
@@ -0,0 +1,186 @@
+                n_len = len(distances)
+                # potentially add if n_len == 0: print (increase radius)
+                self.all_n_index[i][:n_len] = n_index
+                self.all_distances[i][:n_len] = distances
+                continue
+            # removing extra values so the length is equal to max_neigh
+            # values are sorted by distance so only nearest neighbors are kept
+            elif len(distances) > self.max_neigh:
+                # this sorts the list min -> max and returns the indicies
+                sorted_dist_i = np.argsort(distances)
+                self.all_n_index[i] = n_index[sorted_dist_i[: self.max_neigh]]
+                self.all_distances[i] = distances[
+                    sorted_dist_i[: self.max_neigh]
+                ]
+
+    def _gen_features(self):
+        # expand distances in gaussian basis
+        gaussian_distances = expand_distance_gauss(self.all_distances)
+        # reshape the array
+        self.gauss_distances = gaussian_distances.reshape(
+            (
+                gaussian_distances.shape[0] * gaussian_distances.shape[1],
+                gaussian_distances.shape[2],
+            )
+        )
+        # convert to torch tensor, gauss_distances == edge_attr
+        self.edge_attr = torch.FloatTensor(self.gauss_distances)
+        # One-hot encoding for atom type embeddings
+        embeddings = np.vstack(
+            [EMBEDDINGS[site.specie.number] for site in self.struct]
+        )
+        # conver to torch tensor
+        self.embeddings = torch.Tensor(embeddings)
+        # edge_index
+        self.all_n_index = self.all_n_index.reshape(
+            (self.all_n_index.shape[0] * self.all_n_index.shape[1],)
+        )
+        self.edge_index = torch.LongTensor(
+            [self.all_c_index, self.all_n_index]
+        )
+
+    def _clear(self):
+        self.atoms = None
+        self.struct = None
+        self.embedding = None
+        self.edge_index = None
+        self.edge_attr = None
+
+    def convert(
+        self,
+        atoms_collection,
+        processed_file_path=None,
+        collate_and_save=False,
+    ):
+
+        if isinstance(atoms_collection, list):
+            atoms_iter = atoms_collection
+        elif isinstance(atoms_collection, ase.db.sqlite.SQLite3Database):
+            atoms_iter = atoms_collection.select()
+
+        for atoms in tqdm(
+            atoms_iter,
+            desc="converting ASE atoms collection to graphs",
+            total=len(atoms_collection),
+            unit=" systems",
+        ):
+            # check if atoms is an ASE Atoms object this for the ase.db case
+            if not isinstance(atoms, ase.atoms.Atoms):
+                atoms = atoms.toatoms()
+            energy = atoms.get_potential_energy()
+            self._get_neighbors_pymatgen(atoms)
+            self._pad_arrays()
+            self._gen_features()
+
+            # put data in torch geometric format
+            data = Data(
```

Comment:
This is limited to energies, see [this](https://github.com/Open-Catalyst-Project/baselines/blob/acdac7cb8c0f235f7421ad277a0194c2bdb3aa28/ocpmodels/datasets/co_cu_md.py#L110) for what's needed for forces.

We should make this robust for energy+forces. Or possibly not even define the torch geometric here and allow specific datasets to construct them as they wish.